### PR TITLE
[ready] Fix move backward

### PR DESCRIPTION
**Pull request by fnrizzi**
this PR fixes a bug in how the indexing was done in the impl of `move_backward`. It also adds a test as it seems the was not one before (which is strange).

### CODE REVIEW

**Comment by fnrizzi** [comment]
retest this

**Comment by fnrizzi** [comment]
unrelated failure

File: algorithms/unit_tests/TestStdAlgorithmsMoveBackward.cpp [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+namespace KE = Kokkos::Experimental;
+
+template <class Tag, class ValueType, class InfoType>
+void run_single_scenario(const InfoType& scenario_info, int apiId) {
+  const std::size_t view_ext = std::get<1>(scenario_info);
+
+  auto v = create_view<ValueType>(Tag{}, view_ext, "v");
+
+  // v might not deep copyable so to modify it on the host
+  // need to do this
```
Comment (cz4rs) [inline algorithms/unit_tests/TestStdAlgorithmsMoveBackward.cpp:62]:
```suggestion
  // v might be not deep copyable so to modify it on the host
  // we need to do this
```

File: algorithms/unit_tests/TestStdAlgorithmsMoveBackward.cpp [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  auto v_dc   = create_deep_copyable_compatible_view_with_same_extent(v);
+  auto v_dc_h = create_mirror_view(Kokkos::HostSpace(), v_dc);
+  Kokkos::Random_XorShift64_Pool<Kokkos::DefaultHostExecutionSpace> pool(12371);
+  Kokkos::fill_random(v_dc_h, pool, 0, 523);
+  // copy to v_dc and then to v
+  Kokkos::deep_copy(v_dc, v_dc_h);
+  CopyFunctor<decltype(v_dc), decltype(v)> F1(v_dc, v);
+  Kokkos::parallel_for("copy", v.extent(0), F1);
+
+  // make a gold copy of v right now because we call the algorithm
```
Comment (cz4rs) [inline algorithms/unit_tests/TestStdAlgorithmsMoveBackward.cpp:72]:
```suggestion
  // make a gold copy of v before we call the algorithm
```

File: algorithms/unit_tests/TestStdAlgorithmsMoveBackward.cpp [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  } else if (apiId == 1) {
+    auto rit       = KE::move_backward("mylabel", exespace(), KE::begin(v),
+                                 KE::end(v), KE::end(v2));
+    const int dist = KE::distance(KE::begin(v2), rit);
+    EXPECT_TRUE(dist == 5);
+  } else if (apiId == 2) {
+    auto rit       = KE::move_backward(exespace(), v, v2);
+    const int dist = KE::distance(KE::begin(v2), rit);
+    EXPECT_TRUE(dist == 5);
+  } else if (apiId == 2) {
```
Comment (cz4rs) [inline algorithms/unit_tests/TestStdAlgorithmsMoveBackward.cpp:93]: `apiId == 2` is already covered in line 89
```suggestion
  } else if (apiId == 3) {
```

File: algorithms/unit_tests/TestStdAlgorithmsMoveBackward.cpp (hunk 1 above)
Comment (fnrizzi) [reply algorithms/unit_tests/TestStdAlgorithmsMoveBackward.cpp:62]: fixed

File: algorithms/unit_tests/TestStdAlgorithmsMoveBackward.cpp (hunk 3 above)
Comment (fnrizzi) [reply algorithms/unit_tests/TestStdAlgorithmsMoveBackward.cpp:93]: good catch

File: algorithms/unit_tests/TestStdAlgorithmsMoveBackward.cpp [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+  Kokkos::parallel_for("copy", v.extent(0), F1);
+
+  // make a gold copy of v right now because we call the algorithm
+  auto gold = create_host_space_copy(v);
+
+  // create another view that is bigger than v
+  // because we need it to test the move_backward
+  auto v2 = create_view<ValueType>(Tag{}, view_ext + 5, "v2");
+
+  if (apiId == 0) {
```
[the remark this answers is not in the document]
Comment (fnrizzi) [reply algorithms/unit_tests/TestStdAlgorithmsMoveBackward.cpp:79]: good suggestion, but that is not a major gain from using int, is it?

**Comment by fnrizzi** [comment]
unrelated failure from OpenMPTarget

**Review by PhilMiller — APPROVED** [review]

File: algorithms/unit_tests/TestStdAlgorithmsMoveBackward.cpp [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    auto rit =
+        KE::move_backward(exespace(), KE::begin(v), KE::end(v), KE::end(v2));
+    const int dist = KE::distance(KE::begin(v2), rit);
+    EXPECT_TRUE(dist == 5);
```
[the remark this answers is not in the document]
Comment (fnrizzi) [reply algorithms/unit_tests/TestStdAlgorithmsMoveBackward.cpp:84]: done!

**Pull request merged**